script.js

```diff
@@ -0,0 +1,406 @@
+    },
+    arrowup: () => {
+      inputArea.setCaretPosition(0);
+    },
+    arrowdown: () => {
+      inputArea.setCaretPosition(inputArea.value.length);
+    },
+  };
+  const keyCode = parseInt(element.dataset.key, 10);
+  const position = inputArea.getCaretPosition();
+  moveTo[getKeyByValue(keyCode)](position);
+}
+
+function keyCapsLockClick() {
+  if (capsPressed) {
+    document.querySelectorAll('.key--capslock').forEach((key) => {
+      key.removeAttribute('data-selected');
+    });
+    capsPressed = false;
+  } else {
+    document.querySelectorAll('.key--capslock').forEach((key) => {
+      key.setAttribute('data-selected', 'on');
+    });
+    capsPressed = true;
+  }
+  toggleLettersUpperCase();
+}
+
+function keyBackSpaceClick() {
+  const start = inputArea.selectionStart;
+  if (inputArea.hasSelection()) {
+    inputArea.insertValue(start, inputArea.selectionEnd);
+  } else {
+    inputArea.insertValue(start - 1, inputArea.getCaretPosition());
+  }
+}
+
+function keyClick(element) {
+  if (element.classList.contains('key--capslock')) {
+    keyCapsLockClick();
+  }
+  if (element.classList.contains('key--backspace')) {
+    keyBackSpaceClick();
+  }
+  if (element.classList.contains('key--shift')) {
+    toggleLettersUpperCase();
+  }
+  if (element.classList.contains('key--enter')) {
+    inputArea.setValue('\n');
+  }
+  if (element.classList.contains('key--tab')) {
+    inputArea.setValue('\t');
+  }
+  if (element.classList.contains('key--space')) {
+    inputArea.setValue(' ');
+  }
+  if (element.classList.contains('key--letter')) {
+    keyLetterClick(element);
+  }
+  if (element.classList.contains('key--double')) {
+    keyDoubleClick(element);
+  }
+  if (element.classList.contains('key--arrow')) {
+    keyArrowClick(element);
+  }
+  return true;
+}
+
+function keyClickEventHandler(e) {
+  e.preventDefault();
+  if (e.button !== 2) {
+    const keyTarget = e.target.closest('[data-key]');
+    if (keyTarget) {
+      keyClick(keyTarget);
+    }
+  }
+}
+
+function init() {
+  document.body.innerHTML = descriptionHtml;
+  inputArea = document.createElement('textarea', { is: 'kb-textarea' });
+  inputArea.name = 'post';
+  inputArea.cols = 50;
+  inputArea.rows = 5;
+  inputArea.className = 'textarea';
+  inputArea.addEventListener('blur', (e) => e.target.focus());
+  inputArea.autofocus = true;
+  document.body.append(inputArea);
+}
+
+function buildKeyboardLayouts() {
+  keyboards.englishLayout = document.createElement('div');
+  keyboards.englishLayout.className = `keyboard keyboard_en ${currentLanguage === 'en' ? 'on' : ''}`;
+  keyboards.russianLayout = document.createElement('div');
+  keyboards.russianLayout.className = `keyboard keyboard_ru ${currentLanguage === 'ru' ? 'on' : ''}`;
+  const wrapper = document.createElement('div');
+  wrapper.className = 'wrapper';
+  wrapper.appendChild(keyboards.englishLayout);
+  wrapper.appendChild(keyboards.russianLayout);
+  document.body.append(wrapper);
```

Comment:
you can move creating a language layout into a separate function and pass only the 'en' or 'ru'